src/molecules/BottomSheet/BottomSheet.native.js

```diff
@@ -0,0 +1,190 @@
+  (
+    {
+      snapPoint = DEFAULT_SNAP_POINT,
+      children,
+      onBackDropPress = () => {},
+      onBackButtonPress = () => {},
+      onOpened = () => {},
+      onClosed = () => {},
+      adjustToContentHeight = false,
+      alwaysOpen = 0,
+    },
+    ref,
+  ) => {
+    const theme = useTheme();
+    const [headerHeight, setHeaderHeight] = useState(0);
+    const [contentHeight, setContentHeight] = useState(0);
+    const bottomsheetChildrenGroupByType = reactChildrenGroupByType(children);
+
+    const headerComponent = bottomsheetChildrenGroupByType[Header];
+    const footerComponent = bottomsheetChildrenGroupByType[Footer];
+    const contentComponent = bottomsheetChildrenGroupByType[Content];
+
+    if (headerComponent?.length > 1) {
+      throw new Error(
+        `expected to have single \`BottomSheet.Header\` but found ${headerComponent.length}`,
+      );
+    }
+
+    if (footerComponent?.length > 1) {
+      throw new Error(
+        `expected to have single \`BottomSheet.Footer\` but found ${footerComponent.length}`,
+      );
+    }
+
+    const handleHeaderLayoutChange = useCallback((e) => {
+      setHeaderHeight(e.nativeEvent.layout.height);
+    }, []);
+
+    const handleContentLayoutChange = useCallback((e) => {
+      setContentHeight(e.nativeEvent.layout.height);
+    }, []);
+
+    let contentContainerHeight = DEFAULT_SNAP_POINT - headerHeight;
+    if (alwaysOpen > 0) {
+      contentContainerHeight = alwaysOpen - headerHeight;
+    }
+    const isScrollableContent = contentHeight > contentContainerHeight;
+
+    return (
+      <RNModalize
+        ref={ref}
+        snapPoint={snapPoint}
+        HeaderComponent={
+          <View onLayout={handleHeaderLayoutChange}>
+            <HeaderContainer>
+              <Flex alignItems="center">
+                <Space padding={[1, 0, 1.5, 0]}>
+                  <View>
+                    <Size height={0.5} width={8}>
+                      <BottomSheetDragBar />
+                    </Size>
+                  </View>
+                </Space>
+              </Flex>
+              {headerComponent?.length ? (
+                <>
+                  {headerComponent}
+                  <Divider color="shade.920" horizontal />
+                </>
+              ) : null}
+            </HeaderContainer>
+          </View>
+        }
+        FloatingComponent={
+          <Position position="absolute" left={0} right={0} bottom={0}>
+            <View>
+              {isScrollableContent && (
+                <LinearGradient
+                  locations={linearGradientLocations}
+                  colors={[
+                    'rgba(249, 251, 254, 1)',
+                    'rgba(249, 251, 254, 0.6)',
+                    'rgba(249, 251, 254, 0)',
+                  ]}
+                  useAngle={true}
+                  angle={180}
+                  style={styles.linearGradient()}
+                >
+                  <Size height={7}>
```

Comment:
what is this mystery? 🤔

Reply:
<img width="565" alt="Screenshot 2020-07-16 at 4 47 21 PM" src="https://user-images.githubusercontent.com/56016210/87665134-10d18b00-c784-11ea-83f1-7db771d6adf7.png">
This is the view to which gradient is applied.

Reply:
Can't you apply `Size` to this LinearGradient directly? like this

```jsx
<Size height={7}>
  <LinearGradient></LinearGradient>
</Size>
```